next-cloudinary/src/components/CldUploadWidget/CldUploadWidget.types.ts

```diff
@@ -33,24 +66,13 @@ export interface CldUploadWidgetProps {
   uploadPreset?: string;
 }
 
-export interface CldUploadWidgetPropsChildren {
+export interface CldUploadWidgetPropsChildren extends CldUploadWidgetInstanceMethods {
```

Comment:
i'm not a typescript expert by any means, but would the & syntax be more appropriate here? in the event that we'd want to extend multiple type definitions here?

i think that requires it to be a `type`, but just thinking out loud if for instance in the future we'd want to do something like 

```
type CldUploadWidgetPropsChildren = CldUploadWidgetInstanceMethods & SomethingElse & {
```

this isn't urgent / necessary, but thought id bring it up

Reply:
So that is possible with interfaces as well:
```
export interface CldUploadWidgetPropsChildren extends CldUploadWidgetInstanceMethods, SomethingElse { 
```
only difference being how [conflicts gets handled](https://www.typescriptlang.org/docs/handbook/2/objects.html#interfaces-vs-intersections), with `extends` you can't have properties of same name in the interface/type you are extending.
But I can change it to `type` and use `Intersection` operator as well if we want to prefer that.

Reply:
no preference! im still somewhat new to Typescript, so ill defer to your judgement there :)

Reply:
Although, there is no much difference between both the approaches, but I think type's `Intersection` is relatively more used than `extends` approach. Hence, let use that only.